### DISCUSSION TITLE
Webhook sends two requests for check_suite.requested on commit

### DISCUSSION THREAD

**Discussion by xpert98** [community/community#24364] [Apps, API and Webhooks]
<p>Upon commit, my GitHub App gets two requests for check_suite.requested (same check_suite.id).  I’m at a loss to understand why there are two separate requests for the same ID.  Because the App gets two of these, the App creates two check runs with different IDs but only one check run is desired.</p>
<p>Does anybody know why there are two requests sent for check_suite.requested?</p>

**Comment by francisfuzz** [comment]
<aside class="quote no-group" data-username="xpert98" data-post="1" data-topic="116891">
<div class="title">
<div class="quote-controls"></div>
 xpert98:</div>
<blockquote>
<p>Does anybody know why there are two requests sent for check_suite.requested?</p>
</blockquote>
</aside>
<p>:wave: <a class="mention" href="/xpert98">@xpert98</a> –– welcome to GitHub’s Support Community! Thanks for posting your question about these webhooks in relation to check suite activity.</p>
<p>I’m interested in seeing what you’re seeing. :eyes: Given a new GitHub App, could you please share the configuration and the exact steps taken to reliably reproduce those two <code>check_suite.requested</code> events? Also, are you able to share an example of this particular case in the form of a public repository? That would help us investigate further! :bowing_man:</p>

**Comment by xpert98** [comment]
<p>The repo I’m using to test out this app (“proton probot app”) is <a href="https://github.com/xpert98/proton" rel="nofollow noopener">https://github.com/xpert98/proton</a>.</p>
<p>I’ve installed my app and have it set to have the following Repository permissions:</p>
<ul>
<li>Actions: No access</li>
<li>Administration: No access</li>
<li>Checks: Read &amp; write</li>
<li>Content references: No access</li>
<li>Contents: Read-only</li>
<li>Deployments: No access</li>
<li>Issues: No access</li>
<li>Metadata: Read-only</li>
<li>Pages: No access</li>
<li>Pull requests: Read-only</li>
<li>Webhooks: Read &amp; write</li>
<li>Projects: No access</li>
<li>Secrets: No access</li>
<li>Security events: No access</li>
<li>Single file: No access</li>
<li>Commit statuses: Read-only</li>
<li>Dependabot alerts: No access</li>
<li>Workflows: No access</li>
</ul>
<p>It has no Org or User permissions.</p>
<p>Lastly, it is only subscribed to Check suite events.</p>

**Comment by xpert98** [comment, marked as answer]
<p>After a lot of troubleshooting, this appears to only happen when proxying through smee!  I deployed my app to AWS and it consistently receives only one check_suite request.</p>

**Comment by francisfuzz** [comment]
<aside class="quote no-group" data-username="xpert98" data-post="4" data-topic="116891" data-full="true">
<div class="title">
<div class="quote-controls"></div>
 xpert98:</div>
<blockquote>
<p>After a lot of troubleshooting, this appears to only happen when proxying through smee!</p>
</blockquote>
</aside>
<p><a class="mention" href="/xpert98">@xpert98</a> - Thanks for sharing that detail here! It’s great to know that you’ve isolated it down to <code>smee</code>. If you’ve found this behavior to be reliably reproducible, that particular case may help the Probot team investigate further since they maintain the <code>smee.io</code> project:</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/probot/smee-client/issues" target="_blank" rel="noopener">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/15894826/181084358-2f96564e-8ccf-4414-9264-a4a6fbe0b3a4.png" class="thumbnail onebox-avatar" width="400" height="400">

<h3><a href="https://github.com/probot/smee-client/issues" target="_blank" rel="noopener">probot/smee-client</a></h3>

<p>🔴 Receives payloads then sends them to your local server - probot/smee-client</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
